solidity/contracts/utility/MathEx.sol

```diff
@@ -34,6 +38,42 @@ library MathEx {
         return x * x == _num ? x : x + 1;
     }
 
+    /**
+     * @dev computes a powered ratio
+     *
+     * @param _n   ratio numerator
+     * @param _d   ratio denominator
+     * @param _exp ratio exponent
+     *
+     * @return powered ratio's numerator and denominator
+     */
+    function poweredRatio(
+        uint256 _n,
+        uint256 _d,
+        uint256 _exp
+    ) internal pure returns (uint256, uint256) {
```

Comment:
This can be a good opportunity to convert the whole contract (or only the new function) to the new coding conventions we're using with new code